### PR TITLE
clean up build (including a fix to make LazyListGCTest run)

### PR DESCRIPTION
**Pull request by SethTisue**
thank you new sbt 1.4 unused-settings warnings for catching the `LazyListGCTest` thing!!

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -66,26 +60,34 @@ lazy val compat = MultiScalaCrossProject(JSPlatform, JVMPlatform, NativePlatform
...
+      Compile / unmanagedSourceDirectories += {
+        val sharedSourceDir = (ThisBuild / baseDirectory).value / "compat/src/main"
         if (scalaVersion.value.startsWith("2.13.")) sharedSourceDir / "scala-2.13"
         else sharedSourceDir / "scala-2.11_2.12"
       },
-      Test / sourceDirectories += (ThisBuild / baseDirectory).value / "compat/src/test/scala-jvm"
     )
     .jvmSettings(
-      junit
+      Test / unmanagedSourceDirectories += (ThisBuild / baseDirectory).value / "compat/src/test/scala-jvm",
```
Comment (NthPortal) [inline build.sbt:70]: it's not clear to me why `sourceDirectories` doesn't work/what it does, but if this fixes it, great

File: build.sbt [hunk 2]
```diff
@@ -264,14 +266,6 @@ lazy val `scalafix-tests` = project
   .dependsOn(`scalafix-input`, `scalafix-rules`)
   .enablePlugins(BuildInfoPlugin, ScalafixTestkitPlugin)
 
-lazy val dontPublish = Seq(
-  publishArtifact := false,
-  packagedArtifacts := Map.empty,
-  publish := {},
-  publishLocal := {},
-  scalaModuleMimaPreviousVersion := None
```
Comment (NthPortal) [inline build.sbt:272]: is there going to be an issue with this setting being missing where `don'tPublish` is now replaced by `publish / skip := true`?

**Review by NthPortal — APPROVED** [review]
LGTM, though I am not by any means an sbt expert

File: build.sbt (hunk 1 above)
Comment (SethTisue) [reply build.sbt:70]: same here :-)

File: build.sbt (hunk 2 above)
Comment (SethTisue) [reply build.sbt:272]: before, `scalaModuleMimaPreviousVersion` was set in `commonSettings` build-wide so then we had to replace that with `None`. why I did it that way originally I don't know — just not really thinking, I think?

but in any case, in the new version we only set it in the specific place where we actually want it (namely in `compat`'s `jvmSettings`), and just let it default to `None` everywhere else

**Pull request merged**